docs/yaml/functions/find_program.yaml

```diff
@@ -5,7 +5,9 @@ description: |
   to be searched for in `PATH` or other places inside the project.
   The search order is:
 
-  1. Program overrides set via [[meson.override_find_program]]
+  1. Program overrides set via [[meson.override_find_program]]. That is, a
+     subproject includes `meson.override_find_program('myprog')` in its Meson 
+     build file.
```

Comment:
Hmm, "that is" implies that it is only valid like this.

Maybe instead we should say:

```
Program overrides set via [[meson.override_find_program]], e.g. in a previously evaluated subproject's meson.build file.
```

This also makes it clearer that it's not the same as option 7 -- this option 1 is specifically about cases where override_find_program has already run, it's just that we usually expect that to be useful in a subproject.